nuxeo-features/nuxeo-platform-userworkspace/nuxeo-platform-userworkspace-core/src/main/java/org/nuxeo/ecm/platform/userworkspace/core/service/AbstractUserWorkspaceImpl.java

```diff
@@ -228,7 +229,25 @@ protected PathRef getExistingUserWorkspaceRoot(CoreSession session, String usern
         if (session.exists(rootref)) {
             return rootref;
         }
-        return new PathRef(new UnrestrictedRootCreator(rootref, username, session).create().getPathAsString());
+
+        DocumentModel uwsRootModel = doCreateUserWorkspacesRoot(session, rootref);
+
+        DocumentModel uwsRootDoc = CoreInstance.doPrivileged(session, s -> {
+            DocumentModel docModel;
+            try {
+                docModel = s.getOrCreateDocument(uwsRootModel, doc -> initCreateUserWorkspaceRoot(s, doc));
+            } catch (DocumentNotFoundException e) {
+                // domain may have been removed !
+                targetDomainName = null;
+                PathRef ref = new PathRef(computePathUserWorkspaceRoot(s, username, null));
+                DocumentModel rootModel = doCreateUserWorkspacesRoot(s, ref);
+                docModel = s.getOrCreateDocument(rootModel, doc -> initCreateUserWorkspaceRoot(s, doc));
+            }
+            docModel.detach(true);
+            return docModel;
```

Comment:
You don't need to detach and you can return `docModel.getPathAsString()` directly.